packages/agreement-process/test/testArchiveAgreement.ts

```diff
@@ -0,0 +1,165 @@
+  getRandomAuthData,
+  randomArrayItem,
+} from "pagopa-interop-commons-test/index.js";
+import { toAgreementV1 } from "../src/model/domain/toEvent.js";
+import {
+  agreementNotFound,
+  agreementNotInExpectedState,
+  operationNotAllowed,
+} from "../src/model/domain/errors.js";
+import { agreementArchivableStates } from "../src/model/domain/validators.js";
+import { addOneAgreement, readLastAgreementEvent } from "./utils.js";
+import {
+  agreementService,
+  agreements,
+  postgresDB,
+} from "./agreementService.integration.test.js";
+
+export function testArchiveAgreement(): void {
+  describe("archive agreement", () => {
+    it("should succeed when the requester is the consumer and the agreement is in an archivable state", async () => {
+      vi.useFakeTimers();
+      vi.setSystemTime(new Date());
+
+      const authData = getRandomAuthData();
+      const eserviceId = generateId<EServiceId>();
+
+      const agreement = getMockAgreement(
+        eserviceId,
+        authData.organizationId,
+        randomArrayItem(agreementArchivableStates)
+      );
+
+      await addOneAgreement(agreement, postgresDB, agreements);
+
+      const agreementId = await agreementService.archiveAgreement(
+        agreement.id,
+        authData,
+        uuidv4()
+      );
+
+      expect(agreementId).toBeDefined();
+      if (!agreementId) {
+        fail("Unhandled error: returned agreementId is undefined");
+      }
+
+      const actualAgreementData: StoredEvent | undefined =
+        await readLastAgreementEvent(agreementId, postgresDB);
+
+      if (!actualAgreementData) {
+        fail("Creation fails: agreement not found in event-store");
+      }
+
+      expect(actualAgreementData).toMatchObject({
+        type: "AgreementUpdated",
+        event_version: 1,
+        version: "0",
+        stream_id: agreementId,
+      });
+
+      const actualAgreement: AgreementV1 | undefined = protobufDecoder(
+        AgreementUpdatedV1
+      ).parse(actualAgreementData.data)?.agreement;
+
+      if (!actualAgreement) {
+        fail("impossible to decode AgreementAddedV1 data");
+      }
+
+      const expectedAgreemenentArchived: Agreement = {
+        ...agreement,
+        state: agreementState.archived,
+        stamps: {
+          ...agreement.stamps,
+          archiving: {
+            who: authData.userId,
+            when: new Date(),
+          },
+        },
+      };
+      expect(actualAgreement).toMatchObject(
+        toAgreementV1(expectedAgreemenentArchived)
+      );
+
+      vi.useRealTimers();
+    });
+
+    it("should throw a agreementNotFound error when the Agreement doesn't exist", async () => {
+      const authData = getRandomAuthData();
+      const eserviceId = generateId<EServiceId>();
+
+      const agreement = getMockAgreement(
+        eserviceId,
+        authData.organizationId,
+        randomArrayItem(agreementArchivableStates)
+      );
+
+      await addOneAgreement(agreement, postgresDB, agreements);
```

Comment:
I created an Agreement to be sure that the error is throw because the requested id is not present and not because the database is empty

Reply:
Makes sense, I'm gonna do the same also in other tests I wrote.
Since you don't need to add the corresponding Eservice to the readmodel, and you just want to have some random agreement persisted, you could simplify this just and just do:

`await addOneAgreement(getMockAgreement(), postgresDB, agreements);`